app/src/main/java/sdp/moneyrun/menu/AddFriendListListAdapter.java

```diff
@@ -0,0 +1,162 @@
+package sdp.moneyrun.menu;
+
+import android.annotation.SuppressLint;
+import android.app.Activity;
+import android.graphics.Color;
+import android.graphics.drawable.GradientDrawable;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import sdp.moneyrun.R;
+import sdp.moneyrun.database.UserDatabaseProxy;
+import sdp.moneyrun.user.User;
+
+public class AddFriendListListAdapter extends ListAdapterWithUser {
+
+    private final int color_light_gray = Color.rgb(220, 220, 220);
+    private final int color_gold = Color.rgb(255,215,0);
+
+    public AddFriendListListAdapter(Activity context, List<User> userList, User user) {
+        super(context,userList, user);
+    }
+
+    @SuppressLint("ViewHolder")
+    public View getView(int position, View view, ViewGroup parent) {
+        view = LayoutInflater.from(getContext()).inflate(R.layout.add_friend_list_item_layout, null, true);
+        User userRequested = getItem(position);
+
+        TextView userNameView = view.findViewById(R.id.add_friend_list_player_name);
+        Button userButtonView = view.findViewById(R.id.add_friend_list_button);
+
+        //Define name
+        userNameView.setText(String.valueOf(userRequested.getName()));
+
+        //Define button, add user id as button tag to retrieve it later
+        userButtonView.setId(position);
+        userButtonView.setTag(R.string.add_friend_tag_1, userRequested.getUserId());
+
+        // Change button given some state: can follow, already followed or invalid
+        if(userRequested.getUserId() == null ||
+                userRequested.getUserId().equals(getCurrentUser().getUserId())){
+            setInvalidButtonType(userButtonView);
+        }else if(getCurrentUser().getFriendIdList().contains(userRequested.getUserId())){
+            userButtonView.setTag(R.string.add_friend_tag_0, true);
+            setButtonType(userButtonView, false);
+        }else{
+            userButtonView.setTag(R.string.add_friend_tag_0, false);
+            setButtonType(userButtonView, true);
+        }
+
+        userButtonView.setOnClickListener(this::defineFollowButton);
+
+        return view;
+    }
+
+    /**
+     * Functionality for every follow/unfollow buttons to add or remove user from the friend list
+     * @param view the button view
+     */
+    private void defineFollowButton(View view){
+        Button button = (Button) view;
```

Comment:
You should make the attribute type a Button so as to avoid problems at runtime and catch them at compile time.

Reply:
done